### PR TITLE
[Feat] Add dashboard page

### LINKED ISSUE #89: [feat] Add Dashboard page
### This issue will be part of ODHack8.0, please apply via Onlydust app
- Create a new page which route will be: /app, this page should be the dashboard page, please do not add the navbar since the navbar will be created separately in another issue.
- Take a look at the figma to see how the dashboard page looks like.  [Figma prototype](https://www.figma.com/design/SCiGViaoPT9UTrT7CsLbAt/Go-Stark-Me?node-id=0-1&t=FUhnIhNcXRIGkQ16-1)
- Add a component called funds which will be the component used to display the funds as cards. Add dummy data to see the actual output of this component.
- Do not worry about functionality yet, we only want to develop the design.
- Knowledge in Nextjs and typescript are recommended.
- Please read [contributors guide](https://github.com/web3wagers/gostarkme/blob/dev/CONTRIBUTORS_GUIDE.md) before asking for an issue.

### PR DESCRIPTION
**Pull request by sanieni6**
# Pull Request

- [✅] Closes #89 

## Changes description

1. Create the dashboard page
2. Create a hook to get the height and width of any component
3. refactor the animation to make it works with the fund card and fix the x-scroling in the root page

Note: the dashboard page is fully responsive

## Current output
1. Dashboard page
image

## Time spent breakdown

1 hour

## Comments

Any other details you might want to share.

### CODE REVIEW

File: frontend/gostarkme-web/components/dashboard/fundCard.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  const [ref, width, height] = useComponentSize();
+  return (
+    <div className="relative" ref={ref}>
+      <div
+        key={index}
+        className="bg-gray-950 shadow-[0px_4px_4px_0px_#00000040] text-white rounded-[10px] py-[32px] px-[24px] md:py-[48px] md:px-[48px] lg:py-[64px] lg:px-[72px] gap-8 md:gap-10 lg:gap-14 flex flex-col items-start justify-between"
+      >
+        <div className="flex flex-col items-start justify-between gap-4 md:gap-6">
+          <p className=" text-sm md:text-base lg:text-lg text-white font-light leading-[22px] md:leading-[25px] lg:leading-[27.6px]">
+            {fund.type} {fund.type === "Project" ? "🚀" : "🎗️"}
```
Comment (adrianvrj) [inline frontend/gostarkme-web/components/dashboard/fundCard.tsx:26]: Could you please use the HTML representation of the emojis:
rocket: U+1F680
heart: U+1FAC0


<br class="Apple-interchange-newline">[U+1FAC0](https://unicode.org/emoji/charts/full-emoji-list.html#1fac0)

**Review by adrianvrj — CHANGES_REQUESTED** [review]

File: frontend/gostarkme-web/components/dashboard/fundCard.tsx (hunk 1 above)
Comment (sanieni6) [reply frontend/gostarkme-web/components/dashboard/fundCard.tsx]: Solved!

File: frontend/gostarkme-web/app/dashboard/page.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      type: "Project",
+      title: "Adrian's fund",
+      description: "I need help with my project",
+    },
+  ];
+  return (
+    <div className="container mx-auto p-6 min-h-screen">
+      <h1 className="text-4xl font-bold mb-6 flex items-center">
+        Latest Funds
+        <span className="ml-2 text-yellow-400">✨</span>
```
Comment (adrianvrj) [inline frontend/gostarkme-web/app/dashboard/page.tsx:42]: Please use the HTML code for this emoji here too 🙏🏻

**Review by adrianvrj — CHANGES_REQUESTED** [review]

File: frontend/gostarkme-web/app/dashboard/page.tsx (hunk 2 above)
Comment (sanieni6) [reply frontend/gostarkme-web/app/dashboard/page.tsx]: Done!

**Review by adrianvrj — APPROVED** [review]

**Comment by adrianvrj** [comment]
LGTM! tysm sir

**Comment by sanieni6** [comment]
My first ever OSS man, thank you so much guys, i am gonna portrait this PR

**Comment by adrianvrj** [comment]
> My first ever OSS man, thank you so much guys, i am gonna portrait this PR

first of many, congrats homie! 🥳

**Pull request merged**